src/kw_env.sh

```diff
@@ -54,6 +58,34 @@ function env_main()
   fi
 }
 
+# If the env option is requested, it means the O= option from the kernel
+# makefile is used, and this option requires a clean kernel tree to work as
+# expected. This function checks if the current kernel tree is clean.
+#
+# Return:
+# Return 22 in case of error
+function validate_env_before_use()
+{
+  local should_fail=0
+  local list_of_object_file
+
+  load_module_text "${KW_ETC_DIR}/strings/env.txt"
+
+  # Check if there is a .config file
+  [[ -f "${PWD}/.config" ]] && should_fail=1
+
+  # Check if there is any object file
+  list_of_object_file=$(find "${PWD}" -name '*.o')
```

Comment:
Don't we need to also check for `.ko` files, `bzImage`, and the like? Maybe if there aren't `.o` files those I mentioned won't be present, but I confess I don't know for sure...

Reply:
Nice catch! I'll update that.